docs/architecture/lb/nginx-os.md

```diff
@@ -0,0 +1,190 @@
+- Without High Availability, an Nginx Open Source load balancer becomes a new single point of failure.
+
+**Please note that Splunk only supports SC4S**. If issues arise due to the load balancer, please reach out to the Nginx support team.
+
+## Install Nginx
+
+1. Refer to the Nginx documentation for instructions on installing Nginx **with the stream module**, which is required for TCP/UDP load balancing. For example, on Ubuntu:
+```bash
+sudo apt update
+sudo apt -y install nginx libnginx-mod-stream
+```
+
+2. (Optionally) Refer to the Nginx documentation for instructions on fine-tuning Nginx performance. For example, you can update the `events` section in your Nginx configuration file:
+
+`/etc/nginx/nginx.conf`
+```conf
+events {
+    worker_connections 20480;
+    multi_accept on;
+    use epoll;
+}
+```
+Please note that actual load balancer fine-tuning is beyond the scope of the SC4S team's responsibility.
+
+## Preserving Source IP
+The default behavior of Nginx is to overwrite the source IP with the LB's IP. While some users accept this behavior, it is recommended to preserve the original source IP of the message.
+
+Nginx offers three methods to preserve the source IP:
+
+| Method                      | Protocol   |
+|-----------------------------|------------|
+| PROXY protocol              | TCP*       |
+| Transparent IP              | TCP/TLS    |
+| Direct Server Return (DSR)  | UDP        |
+
+* TLS PROXY protocol support in SC4S is scheduled for implementation.
+
+Examples for setting up Nginx with the PROXY protocol and DSR are provided below. The Transparent IP method requires complex network configuration. For more details, refer to [this Nginx blog post](https://www.f5.com/company/blog/nginx/ip-transparency-direct-server-return-nginx-plus-transparent-proxy).
+
+
+## Option 1: Configure Nginx Open Source with the PROXY Protocol
+
+### Advantages:
+- Easy to set up
+
+### Disadvantages:
+- Available only for TCP, not for UDP or TLS
+- Overwriting the source IP in SC4S is not ideal; the `SOURCEIP` is a hard macro and only `HOST` can be overwritten
+- Overwriting the source IP is available only in SC4S versions greater than 3.4.0
+
+### Configuration
+
+1. On your load balancer (LB) node, add a configuration similar to the following:
+`/etc/nginx/modules-enabled/sc4s.conf`
+```conf
+stream {
+    # Define upstream for each of SC4S hosts and ports
+    # Default SC4S TCP ports are 514, 601
+    # Include your custom ports if applicable
+    upstream stream_syslog_514 {
+        server <SC4S_IP_1>:514;
+        server <SC4S_IP_2>:514;
+    }
+    upstream stream_syslog_601 {
+        server <SC4S_IP_1>:601;
+        server <SC4S_IP_2>:601;
+    }
+
+    # Define a common configuration block for all servers
+    map $server_port $upstream_name {
+        514   stream_syslog_514;
```

Comment:
Probably we also need tell that we can use port segregation, it will increase performance:

```SC4S_LISTEN_DEFAULT_TCP_PORT=514,515,516,517```